Core/Security/m2crypto/X509Chain.py

```diff
@@ -486,7 +488,7 @@ def generateProxyToFile(self, filePath, lifetime, diracGroup=False, strength=102
     if not retVal['OK']:
       return retVal
     try:
-      with open(filePath, 'w') as fd:
+      with open(filePath, 'wb') as fd:
```

Comment:
I am not sure. When I started using `io` and used text mode, a lot of tests were failing. using `io` was not only warning, but also a recommendation in your "how to migrate PR"

Reply:
That's why I also suggest getting rid of `from io import open`. If you use text mode in Python 2 you have to write `unicode`.

Reply:
Also, particularly in these low level library, I think it makes sense to use things correctly. `bytes` vs `str` are confusing enough when coming from python 2, so now if we start to replace `bytes` with `str` when we can for simplicity, I think it will be even more confusing. So I am more inclined to really use `bytes` if that's what they are now

Reply:
Now, I am probably missing a few `encode/decode` here or there !

Reply:
That's reasonable, I'll deal with them in my PRs after this is merged.